src/Conversion/ONNXToTOSA/DialectBuilder.hpp

```diff
@@ -30,6 +30,30 @@ namespace onnx_mlir {
 // IndexExpr Builder for Shape lowering
 // =============================================================================
 
+struct TosaBuilder : DialectBuilder {
+  TosaBuilder(mlir::Location loc) : DialectBuilder(loc) {}
+  TosaBuilder(mlir::PatternRewriter &b, mlir::Location loc)
+      : DialectBuilder(b, loc), patternRewriter(&b) {}
+  TosaBuilder(const DialectBuilder &db) : DialectBuilder(db) {}
+  virtual ~TosaBuilder() {}
+
+  mlir::Value reshape(mlir::Value &value, llvm::ArrayRef<int64_t> shape);
+  mlir::Value transpose(mlir::Value &value, llvm::ArrayRef<int64_t> perm);
+  template <typename T>
+  mlir::Value getConst(llvm::ArrayRef<T> vec, llvm::ArrayRef<int64_t> shape);
```

Comment:
Is this inspired by some existing implementation? Comparing the general implementation and the specialization for `float` I'm a little concerned that the type assumptions are incorrectly overgeneralized.

If this is indeed an over-generalization, maybe a straightforward overloading in the header file for the supported types would make more sense, and if the implementations can share code via a template then do that in the cpp file?

Reply:
These are the functions from [Ferdinand](https://github.com/ljfitz/onnx-mlir/blob/fcfca0a2cdd358ef04f15ae4804b2bd858374211/src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.cpp#L90-L142) that I just moved into the class. I think they are originally from TF.